### PR TITLE
[finance-report-1681-source-contributions] Publish decision-backed statement contributions

### LINKED ISSUE #1681: [extraction] Publish one versioned source-to-fact contract across live, CSV, and cassette paths
# Nested extraction child of #1834

> Owner: `common/extraction/contract.py` and `common/extraction/readme.md`. This owns extraction's published source-to-fact language, not proof execution or report rendering.

## Root cause

The boundary is split across `StatementSummary`, rows, `extraction_metadata`, and an ephemeral payload. Tests rebuild a weaker statement, recompute closing, force type/currency/confidence, and drop positions. Supported-source truth is separately duplicated in a testing matrix with test paths.

## Solution

Publish two versioned value contracts:

1. `StatementExtractionResult`: source identity/type, institution/account/period, scalar/per-currency balances, confidence/validation, typed transactions/positions, provenance, warnings, review reasons.
2. `SourceCapability`: stable id, intake mode, evidence kinds, review/promotion semantics, produced facts, traceability target, and `supported|manual_trusted|gap`.

Live provider, CSV, and cassette return the same result. #1834 is the sole persistence consumer. Product capabilities reference proof ids, never pytest paths.

## Planned package ACs

- **AC-extraction.result-envelope.1:** complete versioned bank/brokerage result; incompatible versions fail.
- **AC-extraction.result-envelope.2:** live/CSV/cassette conform without implicit type/currency/confidence/closing defaults.
- **AC-extraction.result-envelope.3:** persistence exactly round-trips balances, transactions, positions, confidence, warnings, provenance; source closing is not recomputed.
- **AC-extraction.source-capability.1:** one stable semantic declaration per capability, with no test path.
- **AC-extraction.source-capability.2:** settlement note remains `gap` until lot/fee/dividend facts reach the package; manual-trusted cannot masquerade as parsing.
- **AC-extraction.result-grading.1:** accuracy output names graded fields; ungraded fields cannot inherit the score.

## Acceptance

- [ ] Both contracts are declared package units/interfaces.
- [ ] Live, CSV, cassette, and persisted reload are schema/version equivalent.
- [ ] Brokerage fixtures preserve exact currency, quantity, value, closing, confidence, and provenance.
- [ ] Routing uses typed facts, not test filename/institution defaults.
- [ ] Loss, incompatible defaults, and unknown versions fail before disposition.
- [ ] Test rename does not change product truth.
- [ ] Accuracy separates graded/ungraded fields.

## Dependencies / non-goals

Consumes #1834. #696 owns proof discovery/join. #567 owns source-trust/readiness projection. This does not guarantee novel layouts or auto-promote positions.

## Atomic extraction-package delivery

This contract ships in the single coordinated extraction replacement PR with #1834 and #1483, after #1906 Audit PR-A. It is not a DTO-only preparatory PR.

TraceRecord integration:

- every StatementExtractionResult has a stable id, schema version, producer version, and content/value digest suitable as an immutable TraceRecord target;
- SourceCapability ids are assertion inputs, not runtime success;
- live, CSV, cassette, persistence, and replay emit the same typed target identity without test-owned normalization;
- unknown/incompatible versions are unproven and stop before disposition.

Deletion in the same extraction PR:

- the ephemeral extraction payload and every weaker duplicate envelope;
- implicit institution/test-filename routing and implicit type/currency/confidence/closing defaults;
- testing-owned product capability semantics and pytest-path ownership once #696 consumes SourceCapability ids;
- persistence/reload paths that drop positions, balances, confidence, warnings, provenance, or the source closing value.

The PR is complete only when no extraction consumer can construct a trusted result without the published version/digest.
## Exact deletion contract

Within the coordinated #1834 PR, remove every externally trusted use of tuple returns, `StatementSummary` plus side-list reconstruction, loose `extraction_metadata`, and weak payload adapters. Remove compatibility/default code that derives institution, evidence type, transaction/position currency, confidence, period, or closing value from a test name, route, or current date. Remove persistence/reload projections that omit any published result fact. `StatementSummary` may survive only as a private parser intermediate; it cannot cross persistence, worker, API, review, disposition, or reporting as a competing contract.

Delete testing-owned product capability definitions and pytest-path coupling after #696 consumes stable `SourceCapability` ids. An unknown schema or missing source fact must reach the common result as explicit review/unproven state, never a legacy default.

## 2026-07-18 fidelity lock

The LLM corpus result is evidence only for the facts that survive its own transport. The current cassette seeding pattern is insufficient if it recalculates closing, coerces `SGD` / `BANK_STATEMENT`, or omits positions, original currency, type, confidence, warnings, and provenance. Such normalization can make transaction-only three-statement assertions pass while disproving neither brokerage nor package fidelity.

Add these acceptance locks to the atomic extraction PR:

- [ ] The cassette, CSV, and live adapters all construct the same versioned envelope and preserve every declared field byte-for-value/Decimal-for-Decimal; a mismatch fails before persistence.
- [ ] One brokerage vector with positions and one bank vector with non-default currency/type/confidence traverse result -> review -> disposition -> persisted result without a fixture-side reconstruction.
- [ ] `settlement_note` remains `gap` unless a vector proves statement/PDF or CSV -> lots, fees, dividends -> frozen package; a structured policy fact alone cannot promote it.
- [ ] ESOP, property, and liability remain labelled `manual_trusted` in the capability contract and package manifest; they are not counted as automatic statement parsing.
## 2026-07-18 accepted-fact contribution amendment

A lossless envelope alone is not an authority boundary for non-ledger facts. For each accepted current `StatementExtractionResult`, extraction emits the canonical `TraceRecord` observation and the applicable current decision over its exact version/digest. The decision may authorize only the published source facts; it neither chooses an accounting category nor posts a ledger command.

Positions, closing balances, currencies, confidence, warnings, provenance, and result identity are members of that one target. A position contribution to the package must name the accepted result decision and its exact position refs. A cassette, CSV adapter, review projection, or reporting consumer cannot synthesize a different decision by re-reading a summary or copying a source label.

Additional acceptance locks:

- [ ] The same accepted result target/digest and current decision are observable from live, worker, CSV, cassette, persisted reload, and review paths.
- [ ] Any dropped or reconstructed position/closing/currency/type/confidence field invalidates the source-fact contribution before portfolio/reporting can consume it.
- [ ] A stale, superseded, review-required, or target-mismatched result is an explicit unproven contribution, never a `manual_trusted` or supported-source shortcut.
- [ ] #567 consumes this published contribution shape; it may not scan all `AtomicPosition` rows to discover support or readiness.

### PR DESCRIPTION
**Pull request by wangzitian0**
## Summary
- Publish extraction-owned ResolvedStatementContribution for immutable current source results and their exact decisions.
- Preserve transaction and position facts without exposing extraction ORM rows or reconstructing cassettes.
- Fail closed for missing, stale, cross-tenant, target-mismatched, and non-authoritative source decisions.
- Add extraction package ACs, proof tests, and boundary documentation.

## Verification
- Statement contribution, reviewed envelope, and source-result contracts: 19 passed.
- tools/preflight.py --tier=static passed.
- Commit hooks passed: ruff, ruff-format, mypy, and security checks.

Closes #1681

### CODE REVIEW

File: apps/backend/src/extraction/extension/statement_contribution.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    if source_record is None or source_record.user_id != user_id or source_record.statement_id != statement.id:
+        return _unproven(statement_id=statement.id, reason_code="missing_current_source_result")
+    try:
+        source_result = StatementExtractionResult.from_payload(source_record.payload)
+    except (TypeError, ValueError):
+        return _unproven(
+            statement_id=statement.id,
+            source_result_id=source_record.id,
+            reason_code="invalid_current_source_result",
+        )
```
[the remark this answers is not in the document]
Comment (wangzitian0) [reply apps/backend/src/extraction/extension/statement_contribution.py:84]: Fixed in 5caa3bf0. The resolver now treats KeyError from a malformed persisted payload as invalid_current_source_result, and the contribution boundary test proves it returns an unproven result instead of raising.

File: apps/backend/src/extraction/base/contribution.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        ):
+            raise ValueError("an authoritative statement contribution requires source identity and decision")
+        if (
+            self.effective_period_start is not None
+            and self.effective_period_end is not None
+            and self.effective_period_start > self.effective_period_end
+        ):
+            raise ValueError("effective statement period is invalid")
+        if self.state == "unproven" and not self.reason_code:
+            raise ValueError("an unproven statement contribution requires a reason_code")
```
[the remark this answers is not in the document]
Comment (wangzitian0) [reply apps/backend/src/extraction/base/contribution.py:51]: Fixed in 5caa3bf0. ResolvedStatementContribution now rejects an authoritative reason_code and an unproven decision_id; focused tests cover both invalid combinations.
